frontend/javascripts/oxalis/view/left-border-tabs/layer_settings_tab.tsx

```diff
@@ -457,6 +475,26 @@ class DatasetSettings extends React.PureComponent<DatasetSettingsProps, State> {
             >
               <InfoCircleOutlined />
             </Tooltip>
+            {canBeMadeEditable ? (
+              <Tooltip
+                title="Make this segmentation editable by adding a Volume Annotation Layer."
+                placement="left"
+              >
+                <HoverIconButton
+                  icon={<LockOutlined />}
+                  hoveredIcon={<UnlockOutlined />}
+                  onClick={() => {
+                    const segmentationLayers = getSegmentationLayers(dataset);
+                    const segmentationLayerIndex = segmentationLayers.indexOf(layer);
```

Comment:
I think, there's a bug here, because the `NewVolumeLayerSelection` component uses the index to access into  `const availableSegmentationLayers = _.differenceWith(segmentationLayers, volumeTracingLayers);` (as defined in `AddVolumeLayerModal`). Therefore, there are cases where the index will be incorrect.

To avoid that there are two places which depend on the semantics of the index, I'd rather pass a `preselectedSegmentationLayerName` instead of an `...Index`. Then, the `NewVolumeLayerSelection` component can find the correct index using the name itself.